sdk/storage/azure-storage-blob-cryptography/src/main/java/com/azure/storage/blob/specialized/cryptography/DecryptorV1.java

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.blob.specialized.cryptography;
+
+import com.azure.core.cryptography.AsyncKeyEncryptionKey;
+import com.azure.core.cryptography.AsyncKeyEncryptionKeyResolver;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.core.util.logging.LogLevel;
+import reactor.core.Exceptions;
+import reactor.core.publisher.Flux;
+
+import javax.crypto.Cipher;
+import javax.crypto.NoSuchPaddingException;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+import java.nio.ByteBuffer;
+import java.security.GeneralSecurityException;
+import java.security.InvalidAlgorithmParameterException;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.util.concurrent.atomic.AtomicLong;
+
+import static com.azure.storage.blob.specialized.cryptography.CryptographyConstants.AES;
+import static com.azure.storage.blob.specialized.cryptography.CryptographyConstants.AES_CBC_NO_PADDING;
+import static com.azure.storage.blob.specialized.cryptography.CryptographyConstants.AES_CBC_PKCS5PADDING;
+import static com.azure.storage.blob.specialized.cryptography.CryptographyConstants.ENCRYPTION_BLOCK_SIZE;
+
+class DecryptorV1 extends Decryptor {
+    private static final ClientLogger LOGGER = new ClientLogger(DecryptorV1.class);
+    private LogLevel v1UsageLogLevel;
+
+    protected DecryptorV1(AsyncKeyEncryptionKeyResolver keyResolver, AsyncKeyEncryptionKey keyWrapper,
+        EncryptionData encryptionData) {
+        super(keyResolver, keyWrapper, encryptionData);
+        this.v1UsageLogLevel = LogLevel.WARNING;
+    }
+
+    @Override
+    Flux<ByteBuffer> decrypt(Flux<ByteBuffer> encryptedFlux, EncryptedBlobRange encryptedBlobRange,
+        boolean padding, String requestUri, AtomicLong totalInputBytes, byte[] contentEncryptionKey) {
+        LOGGER.log(this.v1UsageLogLevel, () -> "Downloaded data found to be encrypted with v1 encryption, "
+            + "which is no longer secure. Uri: " + requestUri);
+        this.v1UsageLogLevel = LogLevel.INFORMATIONAL; // Log subsequently at a lower level to not pollute logs
```

Comment:
Two things:
- I'd not lower severity and let users know they have a problem, if they find logs annoying they can always silence them by configuring logger.
- can `requestUri` have SAS ? if yes, we should redact.

Reply:
@alzimmermsft do you have further thoughts on lowering the log level after the first occasion?

Good point about redacting. Do we have utilities for that? I suppose I can construct a URI object and then remove the query since the real value is in the host and path.